src/melee.cpp

```diff
@@ -1870,11 +1870,13 @@ std::vector<special_attack> player::mutation_attacks(Creature &t) const
             }
 
             ///\EFFECT_UNARMED increases chance of attacking with mutated body parts
-
             ///\EFFECT_DEX increases chance of attacking with mutated body parts
-            const int chance = std::max( 1, mut_atk.chance - get_dex() - unarmed );
-            add_msg( m_debug, "%s proc chance: %d", pr.first.c_str(), chance );
-            if( chance != 1 && !one_in( chance ) ) {
+
+            // Calculate actor ability value to be compared against mutation attack difficulty and add debug message
+            const int proc_value = get_dex() + unarmed;
+            add_msg( m_debug, "%s proc chance: %.1f in %.1f", pr.first.c_str(), proc_value, mut_atk.chance );
```

Comment:
`proc_value` is an `int`, using the format "%.1f" for it cause undefined behavior. Use "%d" instead. Same for `mut_atk.chance`, which is also an `int` and does not go well with "%f", use "%d" instead.